server/tests/integration/user.spec.js

```diff
@@ -412,5 +412,173 @@ describe('User Api', function() {
           expect(res.body).to.have.property('lastName', requestBody.lastName)
         })
     })
+
+    it('admins can set other users as admins', async function(){
+      const user = await User.create({
+        firstName: 'first',
+        lastName: 'last',
+        email: '123@example.com',
+        roles: [],
+        provider: 'local',
+        password: '12345678'
+      })
+
+      const adminUser = createTestUser('admin', ADMIN_ROLE)
+      const adminSession = await createUserSession(adminUser)
+      const adminReq = supertest.agent(adminSession.app)
+
+      await adminReq.put('/api/users')
+        .send({
+          _id: user._id,
+          isAdmin: true
+        })
+        .expect(200)
+
+      const updatedUser = await User.findById(user._id).lean()
+      expect(updatedUser).to.have.property('roles')
+      expect(updatedUser.roles).to.include(ADMIN_ROLE)
+    })
+
+    it('regular users can update their own profile', async function(){
+      const user = await User.create({
+        firstName: 'first',
+        lastName: 'last',
+        email: '123@example.com',
+        roles: [],
+        provider: 'local',
+        password: '12345678'
+      })
+
+      const userSession = await createUserSession(user)
+      const userReq = supertest.agent(userSession.app)
+
+      const requestBody = {
+        _id: user._id,
+        created: user.created,
+        displayName: user.displayName,
+        email: 'new.email@example.com',
+        firstName: 'newFirstName',
+        lastName: 'newLastName',
+        provider: user.provider,
+        roles: user.roles,
+        updated: user.updated
+      }
+
+      await userReq.put('/api/users').send(requestBody).expect(200)
+
+      const updatedUser = await User.findById(user._id).lean()
+      expect(updatedUser).to.have.property('email', requestBody.email)
+      expect(updatedUser).to.have.property('firstName', requestBody.firstName)
+      expect(updatedUser).to.have.property('lastName', requestBody.lastName)
+    })
+
+    xit('regular users cannot update another user profile', async function(){
```

Comment:
This test will fail until #347 is fixed so I set it as pending using `xit`.  After #347 is fixed then it can be included.